sdk/core/azure-core-serializer-json-gson/src/main/java/com/azure/core/serializer/json/gson/GsonJsonNull.java

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.serializer.json.gson;
+
+import com.azure.core.util.serializer.JsonNull;
+
+import java.util.Objects;
+
+/**
+ * GSON specific implementation of {@link JsonNull}.
+ */
+public final class GsonJsonNull implements JsonNull {
+    private final com.google.gson.JsonNull jsonNull;
+
+    /**
+     * Constructs a {@link JsonNull} backed by GSON {@link com.google.gson.JsonNull#INSTANCE}
+     */
+    public GsonJsonNull() {
+        this.jsonNull = com.google.gson.JsonNull.INSTANCE;
+    }
+
+    /**
+     * Constructs a {@link JsonNull} backed by the passed GSON {@link com.google.gson.JsonNull}.
+     *
+     * @param jsonNull The backing GSON {@link com.google.gson.JsonNull}.
+     */
+    public GsonJsonNull(com.google.gson.JsonNull jsonNull) {
```

Comment:
if we can have an `INSTANCE` property, do we need this "public" ctr ?

Reply:
Probably not, I added a lot of constructors so cover a bunch of cases when transitioning between external library type and Azure Core types. There are places where these can be reduced, also potentially transitioned to factory methods.